backend/authentication/auth_controller.js

```diff
@@ -150,6 +150,143 @@ exports.githubCallback = async (req, res) => {
   }
 };
 
+//Update User Profile
+exports.updateProfile = async (req, res) => {
+  const { username } = req.params;
+  const { newFullName, newImageUrl, availability, skills, categories } =
+    req.body;
+
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        username: username,
+      },
+    });
+    if (!user) {
+      return res.status(404).json({
+        error: "User not found",
+      });
+    }
+
+    let newProfileData = {};
+
+    if (newFullName) {
+      newProfileData.fullName = newFullName;
+    }
+    if (newImageUrl) {
+      newProfileData.imageUrl = newImageUrl;
+    }
+    if (availability) {
+      newProfileData.availability = availability;
+    }
+
+    const updatedUserProfile = await prisma.user.update({
+      where: {
+        username: username,
+      },
+      data: newProfileData,
+    });
+
+    if (categories && categories.length > 0) {
+      await prisma.user.update({
+        where: {
+          username: username,
+        },
+        data: {
+          categories: {
+            set: [],
+          },
+        },
+      });
+      for (category of categories) {
+        try {
+          await prisma.category.create({
+            data: {
+              category: category,
+              User: {
+                connect: {
+                  username: username,
+                },
+              },
+            },
+          });
+        } catch (error) {
+          res.status(500).json({
+            error: "Error updating categories",
+          });
+        }
+      }
+    }
+    if (skills && skills.length > 0) {
+      await prisma.user.update({
+        where: {
+          username: username,
+        },
+        data: {
+          skills: {
+            set: [],
+          },
+        },
+      });
+
+      for (skill of skills) {
+        try {
+          await prisma.skill.create({
+            data: {
+              skill: skill,
+              User: {
+                connect: {
+                  username: username,
+                },
+              },
+            },
+          });
+        } catch (error) {
+          res.status(500).json({
+            error: "Error updating skills",
+          });
+        }
+      }
```

Comment:
same as before let's try use the `createMany `

Reply:
Thanks for the feedback and resource. I just fixed this in a recent commit: https://github.com/izuchukwumba/peered/pull/8/commits/5fe0755ce9404ede218e0bb4bcfe4fbc765b2e99